exporter/exporterhelper/internal/queuebatch/persistent_queue.go

```diff
@@ -238,6 +243,83 @@ func (pq *persistentQueue[T]) Offer(ctx context.Context, req T) error {
 	return pq.putInternal(ctx, req)
 }
 
+type spanContextConfigWrapper struct {
+	TraceID    string
+	SpanID     string
+	TraceFlags string
+	TraceState string
+	Remote     bool
+}
+
+type spanContext trace.SpanContext
```

Comment:
Can we wrap it in another context struct, so we can add metadata keys lated without breaking it?